test/governance/ATMTokenTest.js

```diff
@@ -0,0 +1,190 @@
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to mint correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                let result;
+                await instance.mintVesting(receipent, amount, vestingPeriod, { from: daoAgent });
+                if (receipent == daoMember1) {
+                    result = await instance.isVested(daoMember2);
+                } else {
+                    result = await instance.isVested(receipent);
+                }
+                // Assertions
+                assert(!mustFail, 'It should have failed because the amount is greater than the cap');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _8_set_supply_cap_basic: [70000, daoAgent, undefined, false],
+        _9_set_supply_cap_invalid_sender: [100000, daoMember1, 'PauserRole: caller does not have the Pauser role', true]
+    },function(
+        newCap,
+        sender,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to set cap correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.setCap(newCap, { from: sender });
+                const result = await instance.cap();
+                // Assertions
+                assert(!mustFail, 'It should have failed because the sender is invalid');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _10_claim_vested_basic: [daoMember2, 1000, 7000, 7001, undefined, false],
+        _11_claim_vested_before_deadline: [daoMember2, 1000, 7000, 5000, 'Vesting deadline has not passed', true],
+        _12_claim_vested_no_amount: [daoMember1, 1000, 7000, 8000, 'Account does not have a vesting balance!', true]
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        claimTime,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to claim correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.mintVesting(daoMember2, amount, vestingPeriod, { from: daoAgent });
```

Comment:
I am not sure what is the function you are testing:
- mintVesting or
- withdrawVested

After the assert(result) you need to add assertions, if not it only tests the function doesn't fail. Maybe assert balance (you are withdrawing) and event.

Reply:
Added assertions. Was testing the withdrawVested